### PR TITLE
JVM_IR: use substituted return type in function reference invoke

### PR DESCRIPTION
**Pull request by pyos**
#KT-46982 Fixed

### CODE REVIEW

**Comment by dnpetrov** [comment]
LGTM. I'll run tests and push.

**Comment by dnpetrov** [comment]
5c2753b5d141f754f37f510b341f986fb3ead9c8

File: compiler/testData/codegen/box/callableReference/arrayConstructor.kt [hunk 1]
```diff
@@ -0,0 +1,6 @@
+// WITH_RUNTIME
+// IGNORE_BACKEND: JS_IR
```
Comment (romanart) [inline compiler/testData/codegen/box/callableReference/arrayConstructor.kt:2]: 🤔

Comment (pyos) [reply compiler/testData/codegen/box/callableReference/arrayConstructor.kt:2]: Same error as in my other PR related to array constructors - https://youtrack.jetbrains.com/issue/KT-46520

**Pull request closed without merging**